beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -632,6 +637,37 @@ pub fn signature_verify_chain_segment<T: BeaconChainTypes>(
         &chain.spec,
     )?;
 
+    // Verify signatures before matching blocks and data. Otherwise we may penalize blob or column
+    // peers for valid signatures if the block peer sends us an invalid signature.
+    let pubkey_cache = get_validator_pubkey_cache(chain)?;
```

Comment:
Can we put this in a block or add an explicit `drop` for the `pubkey_cache` read guard?

We previously had `drop(pubkey_cache);`

Reply:
done